src/Nodes/Actor.cpp

```diff
@@ -0,0 +1,27 @@
+#include "Nodes/Actor.hpp"
+
+Actor::Actor(ActorRaceEnum raceIn, std::shared_ptr<ActorBehaviour> behav)
+    : race(raceIn), behaviour(behav)
+{
+    // przykładowa inicjalizacja
+    stats["HP"]     = 100;
+    stats["Attack"] = 10;
+}
+
+void Actor::update(float delta) {
+    if (behaviour)
+        behaviour->behave(*this);
+}
+
+void Actor::changeStat(const std::string& name, int delta) {
+    // istnienie klucza:
+    auto it = stats.find(name);
+    if (it == stats.end()) {
+        // rzucamy wyjątek, jeśli statystyka o podanej nazwie nie istnieje
+        throw std::out_of_range("Stat '" + name + "' does not exist");
```

Comment:
I would make it a std::runtime_error, but isn't exactly wrong. This should propably be a warning though. If you start implementing events please aslo include a class that changed the stat into the warning message, something like
```suggestion
        Logger::log(Logger::MessageType::Warning, caller, 
                           " tried changing non existant stat ", name);
```